async/deferred.ts

```diff
@@ -6,6 +6,8 @@
 // typescript produces broken code when targeting ES5 code.
 // See https://github.com/Microsoft/TypeScript/issues/15202
 // At the time of writing, the github issue is closed but the problem remains.
+
+/** @deprecated (will be removed after 0.214.0) */
```

Comment:
Add the same instruction to use `Promise.withResolvers` here.

Reply:
This is a type? Do we still want to do that?